app/Community/Controllers/MessageController.php

```diff
@@ -38,9 +38,20 @@ public function store(MessageRequest $request): RedirectResponse
                 return back()->withErrors(__('legacy.error.error'));
             }
 
+            foreach ($thread->users as $threadUser) {
+                if (!$threadUser->is($user) && !$user->can('sendToRecipient', [Message::class, $threadUser])) {
+                    return back()->withErrors(__('legacy.error.permissions'));
```

Comment:
This error is misleading. "Insufficient permissions". Something along the lines of "Unable to complete request" might work better.

Reply:
https://github.com/RetroAchievements/RAWeb/pull/2483#discussion_r1631761741